aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -547,12 +548,25 @@ impl AptosVM {
             ..
         } = status
         {
-            let info = module_storage
-                .fetch_module_metadata(module_id.address(), module_id.name())
-                .ok()
-                .flatten()
-                .and_then(|metadata| get_metadata(&metadata))
-                .and_then(|m| m.extract_abort_info(code));
+            let info = if self.features().is_lazy_loading_enabled()
+                && traversal_context
+                    .check_is_special_or_visited(module_id.address(), module_id.name())
+                    .is_err()
+            {
+                // This should never be the case if errors are set correctly. For safety, simply do
```

Comment:
why not log an error though? (error! macro)

Reply:
Is returning None actually safer here though? Just to explain where I'm coming from with a super theoretical scenario / example, but if traversal context were to somehow be corrupted, then this could cause a divergence?

Reply:
Changed to logging + unmetered access, good point